Grant permissions on selected tables, sequences and functions inside a postgresql schema

stissot opened the pull request:
The current `database_postgresql_user` resource can only set perms at the database and schema levels which is not enough to replicate the flexibility and complexity of the [GRANT command](http://www.postgresql.org/docs/9.3/static/sql-grant.html) in PostgreSQL.

So I've added actions and attributes for granting perms on some or all tables, sequences and functions in a schema:
```ruby
    # grant all privileges on all tables, sequences and functions in public schema of foo db
    postgresql_database_user 'foo_user' do
      connection postgresql_connection_info
      database_name 'foo'
      schema_name 'public'
      tables [:all]
      sequences [:all]
      functions [:all]
      privileges [:all]
      action [:grant, :grant_schema, :grant_table, :grant_sequence, :grant_function]
    end ```

SimonKaluza commented:
:+1:  This is a fairly critical feature for PostgreSQL users.

Without it the `database_postgresql_user` LWRP is fairly useless as granting `:all` rights on a schema doesn't by default authorize the user to perform any queries on tables in that schema.

stissot commented:
I've updated the PR from master and added a test case checking that a non-superadmin user that is granted access to a database and schema can view a table.

fabarea commented:
:+1: from my side as well. We face a similar need.

PowerKiKi commented:
This PR was updated so that Travis build passes. The build is actually failing for MySQL, but that has nothing to do with this PR (and [also fails on master](https://travis-ci.org/chef-cookbooks/database/jobs/94070102)). The important thing is that [all PostgreSQL tests pass](https://travis-ci.org/chef-cookbooks/database/builds/99923649).

@tas50 could you please have a look at this and let us know if there is any interest into merging this feature ?

PowerKiKi commented:
Closing this in favor of #213.
